core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -917,43 +832,37 @@ jit_compile_op_br(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
         SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
     }
 
-    return handle_next_reachable_block(cc, p_frame_ip);
+    return true;
 fail:
     return false;
 }
 
+bool
+jit_compile_op_br(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
+{
+    return handle_op_br(cc, br_depth, p_frame_ip)
+           && handle_next_reachable_block(cc, p_frame_ip);
+}
+
 bool
 jit_compile_op_br_if(JitCompContext *cc, uint32 br_depth, uint8 **p_frame_ip)
 {
     JitFrame *jit_frame;
     JitBlock *block_dst;
-    JitReg frame_sp_dst, cond;
+    JitReg cond;
     JitBasicBlock *cur_basic_block, *if_basic_block = NULL;
-    JitValueSlot *frame_sp_src = NULL;
-    JitInsn *insn;
-    bool copy_arities;
-    uint32 offset;
-
-#if 0 /* TODO */
-#if WASM_ENABLE_THREAD_MGR != 0
-    /* Insert suspend check point */
-    if (cc->enable_thread_mgr) {
-        if (!check_suspend_flags(cc, func_ctx))
-            return false;
-    }
-#endif
-#endif
 
     if (!(block_dst = get_target_block(cc, br_depth))) {
         return false;
     }
 
+    /* append IF to current basic block */
     POP_I32(cond);
 
     jit_frame = cc->jit_frame;
     cur_basic_block = cc->cur_basic_block;
     gen_commit_values(jit_frame, jit_frame->lp, jit_frame->sp);
-    clear_values(jit_frame);
+    SET_BB_END_BCIP(cur_basic_block, *p_frame_ip - 1);
```

Comment:
Current block hasn't ended yet, it is br_if but not br, the following opcodes also belongs to current basic block.